### DISCUSSION TITLE
Blocks with nested blocks

### DISCUSSION THREAD

**Discussion by wiesson** [payloadcms/payload#741] [Q&A]
Als already briefly discussed in Discord - is there a way to have blocks that uses blocks? 

```[05:00:23] INFO (payload): Starting Payload...
/Users/wiesson/Dev/tm/cc-payload-cms/node_modules/mongoose/lib/helpers/model/discriminator.js:204
    throw new Error('Discriminator with name "' + name + '" already exists');

Error: Discriminator with name "cta-block" already exists
```

It looks like, they can't occur in another block. I searched my source and there is no second usage of that block. 

I also have seen a collection of blocks example (https://github.com/payloadcms/payload/blob/master/demo/collections/Blocks.ts) but I don't wan't to reuse them (if I understand the purpose of the collection correctly)

// Edit:
My use-case is that I have some base components (e.g. Section) and each Section should have some sub blocks (Call-To-Action buttons, image gallery, "show last of X from collection y", `TextWithImage`.) I want a sweat spot between dynamic and predefined components (blocks) so that the marketing team could quickly add beautiful landing pages. 

If there is another good approach, I'm very interested :)

**Comment by DanRibbens** [comment]
You and James already talked about this on Discord, following up for others to see.

This can now be demonstrated from the Payload repo, we have a new way to run what used to be the demo. Now when you run `yarn dev fields` it will start up an instance with data that shows this working.

Cheers!

**Reply by wiesson** [reply]
Yes! This could be closed :) I'm awaiting the v1 release

**Comment by meshin-dev** [comment]
I have the same issue with existing code base of demo 3.0 after updating from `v3.0.0-beta.125` to `v3.0.0-beta.130`

In `src/collections/Pages/index.ts`:

```
blocks: [CallToAction, Content, MediaBlock, Archive, FormBlock],
```

> Error: Discriminator with name "cta" already exists

Can you suggest how this can be fixed please?

**Reply by meshin-dev** [reply]
It may be related, I added debug:

```
console.log('{{{{{{{{{{{{{{{{{{{BUILDING PAYLOAD CONFIG AGAIN}}}}}}}}}}}}}}}}}}}')

export default buildConfig({
```

And I see in console it appeared twice.
